tests/test_portchannel.py

```diff
@@ -57,3 +57,178 @@ def test_Portchannel(self, dvs, testlog):
+            if route["dest"] == "2.2.2.0/24":
+               found_route = True
+               break
+
+        assert found_route
+
+        # check if route points to next hop group
+        nhg_tbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP_GROUP")
+        (status, fvs) = re_tbl.get(key)
+        for v in fvs:
+            if v[0] == "SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID":
+                nhg_id = v[1]
+
+        (status, fvs) = nhg_tbl.get(nhg_id)
+        assert status
+
+        # check if next hop group consists of 4 members
+        nhg_member_tbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP_GROUP_MEMBER")
+        keys = nhg_member_tbl.getKeys()
+        assert len(keys) == 4
+
+        for key in keys:
+            (status, fvs) = nhg_member_tbl.get(key)
+            for v in fvs:
+                if v[0] == "SAI_NEXT_HOP_GROUP_MEMBER_ATTR_NEXT_HOP_GROUP_ID":
+                    assert v[1] == nhg_id
+
+        # bring PortChannel down
+        dvs.servers[0].runcmd("ip link set down dev eth0")
+        time.sleep(1)
+        ps = swsscommon.ProducerStateTable(self.pdb, "LAG_TABLE")
+        fvs = swsscommon.FieldValuePairs([("admin_status", "up"),("mtu", "9100"),("oper_status", "down")])
+        ps.set("PortChannel001", fvs)
+        time.sleep(1)
+
+        # check if next hop group consists of 3 member
+        keys = nhg_member_tbl.getKeys()
+        assert len(keys) == 3
+
+        # remove IP address
+        tbl = swsscommon.Table(self.cdb, "PORTCHANNEL_INTERFACE")
+        tbl._del("PortChannel001|40.0.0.0/31")
+        tbl._del("PortChannel002|40.0.0.2/31")
+        tbl._del("PortChannel003|40.0.0.4/31")
+        tbl._del("PortChannel004|40.0.0.6/31")
+        time.sleep(1)
+
+        # check application database
+        tbl = swsscommon.Table(self.pdb, "INTF_TABLE:PortChannel001")
+        intf_entries = tbl.getKeys()
+        assert len(intf_entries) == 0
+
+        tbl = swsscommon.Table(self.pdb, "INTF_TABLE:PortChannel002")
+        intf_entries = tbl.getKeys()
+        assert len(intf_entries) == 0
+
+        tbl = swsscommon.Table(self.pdb, "INTF_TABLE:PortChannel003")
+        intf_entries = tbl.getKeys()
+        assert len(intf_entries) == 0
+
+        tbl = swsscommon.Table(self.pdb, "INTF_TABLE:PortChannel004")
+        intf_entries = tbl.getKeys()
+        assert len(intf_entries) == 0
+
+        # remove PortChannel members
+        tbl = swsscommon.Table(self.cdb, "PORTCHANNEL_MEMBER")
+        tbl._del("PortChannel001|Ethernet0")
+        tbl._del("PortChannel002|Ethernet4")
+        tbl._del("PortChannel003|Ethernet8")
+        tbl._del("PortChannel004|Ethernet12")
+        time.sleep(1)
+
+        # remove PortChannel
+        tbl = swsscommon.Table(self.cdb, "PORTCHANNEL")
+        tbl._del("PortChannel001")
+        tbl._del("PortChannel002")
+        tbl._del("PortChannel003")
+        tbl._del("PortChannel004")
+        time.sleep(1)
```

Comment:
Restore eth0 up. If test_port.py follows this test, we got persistent test failure.

Reply:
@wendani , would you please provide the fix?